### DISCUSSION TITLE
Load (download) a component when click a button

### DISCUSSION THREAD

**Discussion by IRediTOTO** [vercel/next.js#16351] [Help]
Hi, i am making edit page :D And yes we can use form. But I want to make something more. 
I want on single product page, if user (owner of that product) click to edit, component input will load from server, not from current code from client or push another page. Is there a feature like that? Thank you.

**Comment by arso1er** [comment]
Hi, dynamic import should help you. Documentation [here](https://nextjs.org/docs/advanced-features/dynamic-import).

**Reply by IRediTOTO** [reply]
thank you for reply, but the component still load, just lazy load.

**Reply by arso1er** [reply]
Maybe I didn't understand your question correctly. Isn't that what you want, ie load the component only when user clicks on edit? 
Additionnaly, you could add `ssr: false`. I used dynamic import with success to reduce bundle size on my current project.

**Reply by IRediTOTO** [reply]
Yes I use it too, but still lazy load :D I think must use pure js to do that.  But ok, i am fine with form
